cfme/tests/pod/test_appliance_crud.py

```diff
@@ -13,29 +18,153 @@
+    # TODO: prepare some serialization functions and add them everywhere
+    tags = json.loads(template['custom_data'].replace("'", '"').replace('u"', '"'))
+    return tags['TAGS']
+
+
+@pytest.fixture
+def temp_pod_appliance(provider, appliance_data):
     with temp_appliances(preconfigured=False, provider_type='openshift',
                          provider=provider.key, template_type='openshift_pod') as appliances:
         with appliances[0] as appliance:
-            appliance.openshift_creds = openshift_creds
+            appliance.openshift_creds = appliance_data['openshift_creds']
             appliance.is_pod = True
             stack.push(appliance)
             yield appliance
             stack.pop()
 
 
+@pytest.fixture
+def temp_pod_ansible_appliance(provider, appliance_data, template_tags):
+    tags = template_tags
+    params = appliance_data.copy()
+    project = 'test-pod-ansible-{t}'.format(t=fauxfactory.gen_alphanumeric().lower())
+    try:
+        with ssh.SSHClient(hostname=params['openshift_creds']['hostname'],
+                           username=params['openshift_creds']['ssh']['username'],
+                           password=params['openshift_creds']['ssh']['password'],
+                           oc_username=params['openshift_creds']['username'],
+                           oc_password=params['openshift_creds']['password'],
+                           project=project,
+                           is_pod=True) as ssh_client:
+
+            # copying ansible configuration file to openshift server
+            fulfilled_config = ansible_config.format(
+                host=provider.provider_data['hostname'],
+                subdomain=provider.provider_data['base_url'],
+                proj=project,
+                app_ui_url=tags['cfme-openshift-app-ui']['url'],
+                app_ui_tag=tags['cfme-openshift-app-ui']['tag'],
+                app_url=tags['cfme-openshift-app']['url'],
+                app_tag=tags['cfme-openshift-app']['tag'],
+                ansible_url=tags['cfme-openshift-embedded-ansible']['url'],
+                ansible_tag=tags['cfme-openshift-embedded-ansible']['tag'],
+                httpd_url=tags['cfme-openshift-httpd']['url'],
+                httpd_tag=tags['cfme-openshift-httpd']['tag'],
+                memcached_url=tags['cfme-openshift-memcached']['url'],
+                memcached_tag=tags['cfme-openshift-memcached']['tag'],
+                db_url=tags['cfme-openshift-postgresql']['url'],
+                db_tag=tags['cfme-openshift-postgresql']['tag'])
+            with tempfile.NamedTemporaryFile('w') as f:
+                f.write(fulfilled_config)
+                f.flush()
+                os.fsync(f.fileno())
+                remote_file = os.path.join('/tmp', f.name)
+                ssh_client.put_file(f.name, remote_file, ensure_host=True)
+
+                # run ansible deployment
+                ansible_cmd = ('/usr/bin/ansible-playbook -v -i {inventory_file} '
```

Comment:
bump this out of the file-handle context as its not working with the file anymore.